src/email_service/health_check/run_repository.py

```diff
@@ -0,0 +1,478 @@
+                try:
+                    dt_obj = datetime.datetime.strptime(value, "%Y-%m-%d %H:%M:%S")
+                except ValueError as e:
+                    logger.error(
+                        "Could not parse timestamp string '%s' for column '%s': %s",
+                        value,
+                        col_name,
+                        e,
+                    )
+                    return value  # Return original string if parsing fails
+            return time_util.format_time_to_iso8601(dt_obj)
+        elif col_name in JSONB_COLUMNS:
+            try:
+                return json.loads(value)
+            except json.JSONDecodeError:
+                return value
+        return value
+    elif "longValue" in field:
+        return field["longValue"]
+    elif "doubleValue" in field:
+        return field["doubleValue"]
+    elif "booleanValue" in field:
+        return field["booleanValue"]
+    elif "blobValue" in field:
+        return field["blobValue"]
+
+    # Handle array types
+    elif "arrayValue" in field:
+        array = field["arrayValue"]
+
+        # Handle various array types
+        if "stringValues" in array:
+            return array["stringValues"]
+        elif "longValues" in array:
+            return array["longValues"]
+        elif "doubleValues" in array:
+            return array["doubleValues"]
+        elif "booleanValues" in array:
+            return array["booleanValues"]
+        elif "arrayValues" in array:
+            # Recursively process nested arrays
+            return [parse_field(col_name, v) for v in array["arrayValues"]]
+        # Empty array
+        return []
+
+    # If type cannot be identified, return the original field
+    return field
+
+
+# Custom JSON encoder to handle Decimal types
+class DecimalEncoder(json.JSONEncoder):
+    """Custom JSON encoder for Decimal objects."""
+
+    def default(self, o: object) -> object:
+        if isinstance(o, Decimal):
+            return float(o)
+        return super().default(o)
+
+
+class RunRepositoryError(Exception):
+    """Run task repository error"""
+
+    def __init__(self, message, sql=None, params=None, original_exception=None):
+        super().__init__(message)
+        self.sql = sql
+        self.params = params
+        self.original_exception = original_exception
+
+
+class RunRepository:
+    """Repository class for managing runs in PostgreSQL"""
+
+    def __init__(self):
+        """
+        Initialize the repository with RDS Data API client.
+        """
+        self._rds_data = boto3.client("rds-data")
+        self._database_name = DATABASE_NAME
+        self._resource_arn = RDS_CLUSTER_ARN
+        self._secret_arn = RDS_CLUSTER_MASTER_USER_SECRET_ARN
+
+    # invoke db
+    def check_connection(self) -> bool:
+        """
+        Select 1 to check database connectivity.
+        """
+        try:
+            sql = "SELECT run_id FROM runs LIMIT 1"
```

Comment:
Sometimes the database may have just been initialized, so there is no data.
I think our intention is to check the connection, not whether the database contains data in this function.
```suggestion
            sql = "SELECT 1"
```